src/state/auctionPrice/hooks.ts

```diff
@@ -0,0 +1,126 @@
+  return useSelector<AppState, AppState['auctionPrice']>((state) => state.auctionPrice)
+}
+
+export function useAuctionPriceHandlers(): {
+  onSetCurrentPrice: (price: string, priceReversed: string) => void
+  onPriceAlteration: () => void
+  onUpdateIfPriceChanged: () => void
+  onResetCurrentPrice: () => void
+} {
+  const dispatch = useDispatch<AppDispatch>()
+
+  const onSetCurrentPrice = useCallback(
+    (price: string, priceReversed: string) => {
+      dispatch(setCurrentPrice({ price, priceReversed }))
+    },
+    [dispatch],
+  )
+  const onPriceAlteration = useCallback(() => {
+    dispatch(alterationCurrentPrice())
+  }, [dispatch])
+  const onUpdateIfPriceChanged = useCallback(() => {
+    dispatch(updateCurrentPrice())
+  }, [dispatch])
+  const onResetCurrentPrice = useCallback(() => {
+    dispatch(resetCurrentPrice())
+  }, [dispatch])
+
+  return {
+    onSetCurrentPrice,
+    onPriceAlteration,
+    onUpdateIfPriceChanged,
+    onResetCurrentPrice,
+  }
+}
+
+function calculateClearingPrice(
+  clearingOrder: Order | null | undefined,
+  auctioningToken: Token | undefined,
+  biddingToken: Token | undefined,
+) {
+  const clearingPriceInfoAsSellOrder = orderToSellOrder(
+    clearingOrder,
+    biddingToken,
+    auctioningToken,
+  )
+  const price = orderToPrice(clearingPriceInfoAsSellOrder)?.toSignificant(5)
+  const priceReversed = orderToPrice(clearingPriceInfoAsSellOrder)?.invert().toSignificant(5)
+
+  return [price, priceReversed]
+}
+
+export function useSetCurrentPrice(
+  auctionIdentifier: AuctionIdentifier,
+  derivedAuctionInfo: DerivedAuctionInfo,
+) {
+  const { shouldLoad: shouldLoadPrice } = useAuctionPriceState()
+  const { onResetCurrentPrice, onSetCurrentPrice } = useAuctionPriceHandlers()
+  const {
+    current: { auctionId, chainId },
+  } = useRef(auctionIdentifier)
+
+  useEffect(() => {
+    onResetCurrentPrice()
+  }, [auctionId, chainId, onResetCurrentPrice])
+
+  useEffect(() => {
+    if (!derivedAuctionInfo || shouldLoadPrice !== PriceStatus.NEEDS_UPDATING) return
+
+    let cancelled = false
+    const fetchApiData = async () => {
+      try {
+        if (!chainId || !auctionId || !additionalServiceApi) {
+          return
+        }
+
+        const clearingOrderAndVolume = await additionalServiceApi.getClearingPriceOrderAndVolume({
+          networkId: chainId,
+          auctionId,
+        })
+
+        if (!cancelled) {
+          const [price, priceReversed] = calculateClearingPrice(
+            clearingOrderAndVolume.clearingOrder,
+            derivedAuctionInfo.auctioningToken,
+            derivedAuctionInfo.biddingToken,
+          )
+          onSetCurrentPrice(price as string, priceReversed as string)
+        }
+      } catch (error) {
+        onResetCurrentPrice()
+        console.error('Error getting clearing price info', error)
+      }
+    }
+
+    fetchApiData()
+    return (): void => {
+      cancelled = true
+    }
```

Comment:
@ramirotw What do you think is the best way to handle this pattern? Maybe checking `fetchApiData().then(() => { if(!cancelled...)` or something like that. Or a hook with support for async functions.